src/solver/variable/economy/bindingConstraints/bindingConstraintsMarginalCost.h

```diff
@@ -0,0 +1,321 @@
+
+                dayInTheYear++;
+            }
+        }
+
+        // For weekly binding constraints, getting weekly marginal price
+        if (associatedBC_->type == Data::BindingConstraint::typeWeekly)
+        {
+            uint weekInTheYear = state.weekInTheYear;
+            double weeklyValue
+              = -state.problemeHebdo->ResultatsContraintesCouplantes[bindConstraintGlobalNumber_]
+                   .variablesDuales[0];
+
+            pValuesForTheCurrentYear[yearMemorySpace_].week[weekInTheYear] = weeklyValue;
+
+            // Even if not printed for weekly BC, we need daily values as : (weekly values) / 7
+            double dailyValue = weeklyValue / 7;
+
+            int dayInTheYear = state.weekInTheYear * 7;
+            for (int dayInTheWeek = 0; dayInTheWeek < 7; dayInTheWeek++)
+            {
+                pValuesForTheCurrentYear[yearMemorySpace_].day[dayInTheYear] = dailyValue;
+                dayInTheYear++;
+            }
+        }
+    }
+
+    void hourBegin(unsigned int hourInTheYear)
+    {
+        // Next variable
+        NextType::hourBegin(hourInTheYear);
+    }
+
+    void hourEnd(State& state, unsigned int hourInTheYear)
+    {
+        if (associatedBC_->type == Data::BindingConstraint::typeHourly)
+        {
+            pValuesForTheCurrentYear[yearMemorySpace_][hourInTheYear]
+              -= state.problemeHebdo->ResultatsContraintesCouplantes[bindConstraintGlobalNumber_]
+                   .variablesDuales[state.hourInTheWeek];
+        }
+
+        NextType::hourEnd(state, hourInTheYear);
+    }
+
+    Antares::Memory::Stored<double>::ConstReturnType retrieveRawHourlyValuesForCurrentYear(
+      unsigned int column,
+      unsigned int numSpace) const
+    {
+        return pValuesForTheCurrentYear[numSpace].hour;
+    }
+
+    void localBuildAnnualSurveyReport(
+      SurveyResults& results,
+      int fileLevel,
+      int precision /* printed results : hourly, daily, weekly, ...*/,
+      unsigned int numSpace) const
+    {
+       
+        // Initializing external pointer on current variable non applicable status
+        results.isCurrentVarNA[0] = precision < pow(2, associatedBC_->type - 1);
+
+        if (AncestorType::isPrinted[0])
+        {
+            // Write the data for the current year
+            results.variableCaption = getBindConstraintCaption();
+            pValuesForTheCurrentYear[numSpace].template buildAnnualSurveyReport<VCardType>(
+              results, fileLevel, precision);
+        }
+    }
+
+private:
+    std::string getBindConstraintCaption() const
+    {
+        return associatedBC_->name + " (" + associatedBC_->operatorType + ")";
+    }
+
+    //! Intermediate values for each year
+    typename VCardType::IntermediateValuesType pValuesForTheCurrentYear;
+    unsigned int pNbYearsParallel;
+    Data::BindingConstraintRTI* associatedBC_ = nullptr;
+    uint yearMemorySpace_ = 0;
+    uint bindConstraintGlobalNumber_ = -1;
```

Comment:
This will lead to invalid reads in arrays. Make sure to check that `bindConstraintGlobalNumber_>=0` before reading e.g `study.runtime->bindingConstraint[bindConstraintGlobalNumber_]`

Reply:
Handled